src/core/agent/services/connectionService.ts

```diff
@@ -77,11 +82,15 @@ class ConnectionService extends AgentService {
       },
     });
 
-    const operation = await this.resolveOobi(url);
-    const connectionId = operation.response.i;
+    const operation = await this.resolveOobi(url, multiSigInvite);
+    const connectionId =
+      operation.done && operation.response
+        ? operation.response.i
+        : new URL(url).pathname.split("/")[2];
     const connectionMetadata: any = {
       alias: operation.alias,
       oobi: url,
+      pending: multiSigInvite ? false : !operation.done,
```

Comment:
I thought we can ignore multisig-related connection for this `pending` status? Because the multi-sig operation can be pending.

Reply:
For now, for a multi-sig we will wait until the operation is done and throw an error if it is not done. For normal connections, after 2 seconds we will set it to the background as an async operation.

What this means is right now for a multi-sig, if we hit this line of code, the operation will always be done. Otherwise we would have thrown an error.

I'm suggesting we just have `pending: !operation.done` because in the future we might want to also allow multi-sigs to use the long running operation tracker (it's just tricky for the UI). If we don't change it now, we might have a bug later if we forget to change this line. So it's safer to do this.

Reply:
Ok, I understand your idea now